ThreadGroup/Controllers/TGRouterAuthViewController.m

```diff
@@ -28,6 +29,7 @@ - (void)viewWillAppear:(BOOL)animated {
 
 - (void)resetView {
     self.errorLabel.hidden = YES;
```

Comment:
You need to re-enable userInteraction here. The user can return to this view and chose another router. The buttons will not function after the first successful joining request.